app/routes.py

```diff
@@ -21,11 +25,63 @@ def upload():
             return render_template('index.html', title='Cheminformatic Analysis', errors=["Please input a valid file format"])
 
         inputs = smiles.keys()
-        
+        global good_smiles
         good_smiles = filter_smiles(pains.get_smiles(inputs))
+        global bad_smiles
+        bad_smiles = pains.get_bad_smiles(inputs)
+        global all_smiles 
+        all_smiles = smiles
+        
+    return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles)
+
+@app.route('/')
+@app.route('/pains_verify_and_coefficient_use')
+def pains_verify_and_coefficient_use():
+  inputs = smiles.keys()
+  global bad_smiles #use the global version 
+  
+  bad_smiles = pains.get_bad_smiles(inputs)
+  
+  print(bad_smiles)
+  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles)
+
+
+@app.route('/verify_pains', methods=['GET', 'POST'])
+def verify_pains():
+  global good_smiles
+  global bad_smiles
+  global all_smiles
+  if request.method == 'POST':
+    form = request.form
+    if form['action'] == 'Drop Selected Compounds':
+      
+      for smile in form:
+        if(smile != 'action'):
+          del bad_smiles[smile]
+    elif form['action'] == 'Ignore Errors':
+      
+      for smile in form:
+        if(smile != 'action'):
+          del bad_smiles[smile]
+          good_smiles[smile] = all_smiles[smile]
+
+  return render_template('pains_verify_and_coefficient_use.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles)
+
+@app.route('/final_compounds', methods=['GET', 'POST'])
+def final_compounds():
+  if request.method == 'POST':
+    tanimoto = request.form['tanimoto']
+  
+  #get real final smiles
+  inputs = smiles.keys()
+  global good_smiles
+  global bad_smiles
+  bad_smiles={}
 
-        tanimoto_smiles = clustering.add_tanimoto_coefficients(good_smiles)
+  tanimoto_smiles = clustering.add_tanimoto_coefficients(good_smiles)
+  
+  tanimoto_smile_strings = [" ".join([str(i) for i in v['similarities'].values()]) for v in tanimoto_smiles.values()]
 
-        get_smiles_json(tanimoto_smiles)
+  get_smiles_json(tanimoto_smiles, float(tanimoto))
 
-        return render_template('cluster.html', title='Cheminformatic Analysis')
+  return render_template('cluster.html', title='Cheminformatic Analysis', bad_smiles=bad_smiles, tanimoto_smiles=tanimoto_smile_strings)
```

Comment:
Since the cluster page no longer displays the smiles or similarities, you can remove the `bad_smiles` and `tanimoto_smiles` parameters